### PR TITLE
update docs of distributed plugin

### PR DESCRIPTION
**Pull request by dovics**

### CODE REVIEW

File: docs/distributed-plugin/README.md [hunk 1]
```diff
@@ -6,12 +6,67 @@
...
+    // The parameter should be SyncedEnforced
+    SetEnforcer(interface{})
+}
+```
+
+
+Since different raft implementations provide many different functions, Dispatcher should only include the functions that casbin-core needs to use, and leave the other parts to the plug-in to implement it by itself. This ensures that users can use the different features of different plug-ins as much as possible
+
+
+The `Propose`  is replaced by `AddPolicy` , `AddPolicies`  etc. The `ConfChange`  is not necessary for casbin-core. It should be defined by the plug-in itself. In addition, there will be other roles in different raft implementations, such as `Learner`  and `NotVoter` . We should not restrict this.
```
Comment (nodece) [inline docs/distributed-plugin/README.md]: There are multiple spaces here.

File: docs/distributed-plugin/README.md [hunk 2]
```diff
@@ -6,12 +6,67 @@
 
 ## Overview
 
-This document outlines the work for how to design the plugin. 
+A plugin to ensure the consistency of multiple casbin instances in distributed situations
 
 ## Motivation
 
-Please visit https://github.com/casbin/Summer2020#3-casbin%E5%88%86%E5%B8%83%E5%BC%8F%E6%8F%92%E4%BB%B6-golang for details.
+Now the distribution of casbin is implemented based on watcher, but watcher is to let all instances reload the database after the update occurs. This is a very resource-consuming operation, so WatcherEx (https://github.com/casbin) /casbin/issues/398), you can notify other instances of the modified content, so as to achieve incremental updates and reduce the number of IOs. However, because Watcher notifies other instances after the update, it cannot be controlled for high concurrency mode, please refer to [https://github.com/casbin/casbin/issues/421](https://github.com/casbin/casbin/issues/421).
```
Comment (nodece) [inline docs/distributed-plugin/README.md]: 1. casbin should be Casbin. 
2. watcher should be **`Watcher`.**
3. the "WatcherEx (https://github.com/casbin) /casbin/issues/398)" seems to be wrong.

File: docs/distributed-plugin/README.md [hunk 3]
```diff
@@ -6,12 +6,67 @@
...
+
+
+In addition, many people hope casbin supports the master-slave mode([https://github.com/casbin/casbin/issues/402).](https://github.com/casbin/casbin/issues/402.) Their main purpose is to ensure data consistency in the case of multiple instances, so raft should be a good choice, this is a protocol that guarantees strong consistency. Through election of Leader, all other nodes are based on Leader to achieve strong consistency of the cluster. This algorithm has been adopted by many large projects such as etcd. Compared with master-slave mode, raft is a more efficient and reliable choice.
```
Comment (nodece) [inline docs/distributed-plugin/README.md]: `([https://github.com/casbin/casbin/issues/402).](https://github.com/casbin/casbin/issues/402.) ` is wrong.

File: docs/distributed-plugin/README.md [hunk 4]
```diff
@@ -6,12 +6,67 @@
...
+
+
+In addition, many people hope Casbin supports the master-slave mode([https://github.com/casbin/casbin/issues/402)](https://github.com/casbin/casbin/issues/402). Their main purpose is to ensure data consistency in the case of multiple instances, so raft should be a good choice, this is a protocol that guarantees strong consistency. Through election of Leader, all other nodes are based on Leader to achieve strong consistency of the cluster. This algorithm has been adopted by many large projects such as etcd. Compared with master-slave mode, raft is a more efficient and reliable choice.
```
Comment (nodece) [inline docs/distributed-plugin/README.md]: please fix the `([https://github.com/casbin/casbin/issues/402)(https://github.com/casbin/casbin/issues/402)` to `[https://github.com/casbin/casbin/issues/402](https://github.com/casbin/casbin/issues/402)` or `(https://github.com/casbin/casbin/issues/402)`.

**Comment by hsluoyz** [comment]
Please add these valuable docs to our official docs site.

**Pull request merged**